portal-api/src/v1/users/routes.js

```diff
@@ -257,34 +268,22 @@ module.exports.resetPasswordWithToken = async (req, res, next) => {
     });
   }
 
+  // Void token - Token expired
   const user = await getUserByPasswordToken(resetPwdToken, req.body);
-
-  if (!user) {
-    return res.status(400).json({
-      success: false,
-      errors: {
-        count: 1,
-        errorList: {
-          currentPassword: {
-            text: 'Password reset link is not valid',
-          },
-        },
-      },
-    });
-  }
-
+  // Stale token - Generated over 24 hours ago
   const hoursSincePasswordResetRequest = user.resetPwdTimestamp
     ? (Date.now() - user.resetPwdTimestamp) / 1000 / 60 / 60
     : 9999;
 
-  if (hoursSincePasswordResetRequest > 24) {
+  // Token check
+  if (!user || (hoursSincePasswordResetRequest > 24)) {
```

Comment:
the token exists in the database for the current user. so we can check if the token that they are using matches the one from the database.